compiler-plugin/src/main/kotlin/arrow/meta/quotes/Parameter.kt

```diff
@@ -0,0 +1,33 @@
+package arrow.meta.quotes
+
+import arrow.meta.Meta
+import arrow.meta.phases.ExtensionPhase
+import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.psi.KtAnnotationEntry
+import org.jetbrains.kotlin.psi.KtParameter
+import org.jetbrains.kotlin.psi.KtTypeParameter
+import org.jetbrains.kotlin.psi.KtTypeReference
+import org.jetbrains.kotlin.psi.psiUtil.modalityModifierType
+import org.jetbrains.kotlin.psi.psiUtil.visibilityModifierType
+
+/**
+ * A [KtParameter] [Quote] with a custom template destructuring [ParameterScope]
+ */
+fun Meta.parameter(
+  match: KtParameter.() -> Boolean,
+  map: ParameterScope.(KtParameter) -> Transform<KtParameter>
+) : ExtensionPhase =
+  quote(match, map) { ParameterScope(it) }
+
+/**
+ * A template destructuring [Scope] for a [KtParameter]
+ */
+class ParameterScope(
```

Comment:
All looks great but this may be missing the default value part of the structure since KtParameters can point to an initializer expression block.

Reply:
Yeah you're right, now I've mapped it as a `KtExpression`, do you think it's enough to go or need some kind of custom handling (mapping different properties of the default value)

Reply:
If the initializer is of type `KtExpression` wrapping it in a `Scope` is all you need to do as that is an ADT we don't yet cover. It includes Constants, Calls and many others we don't yet have as quotes. We can revisit this one once we have those. Thanks so much for this great contribution @bloderxd